common/src/main/java/software/amazon/kms/common/AbstractKmsApiHelper.java

```diff
@@ -40,30 +43,51 @@
         try {
             return serviceCall.get();
         } catch (final AlreadyExistsException e) {
-            throw new CfnAlreadyExistsException(e);
+            throw new CfnAlreadyExistsException(addMessageIfNull(operation, e));
         } catch (final InvalidAliasNameException | KmsInvalidStateException | InvalidArnException |
             MalformedPolicyDocumentException | TagException | UnsupportedOperationException |
             DisabledException e) {
-            throw new CfnInvalidRequestException(e);
+            throw new CfnInvalidRequestException(addMessageIfNull(operation, e));
         } catch (final LimitExceededException e) {
-            throw new CfnServiceLimitExceededException(e);
+            throw new CfnServiceLimitExceededException(addMessageIfNull(operation, e));
         } catch (final InvalidMarkerException e) {
             // We should never make a call with an invalid marker, if we did, there is an issue
-            throw new CfnInternalFailureException(e);
+            throw new CfnInternalFailureException(addMessageIfNull(operation, e));
         } catch (final KmsInternalException | DependencyTimeoutException e) {
             throw new CfnServiceInternalErrorException(operation, e);
```

Comment:
why not do this check here ? Is it only needed when operation is not being based ?

Reply:
I added the check to all of the exception constructors that adopt the exception message of the base exception. CfnServiceInternalErrorException(operation, e) for example, sets the exception message to "Internal error reported from downstream service during operation '$OPERATION'."